lib/puppet/util/tagging.rb

```diff
@@ -1,7 +1,7 @@
 require 'puppet/util/tag_set'
 
 module Puppet::Util::Tagging
-  ValidTagRegex = /\A[0-9A-Za-z_][0-9A-Za-z_:.-]*\Z/
+  ValidTagRegex = /\A[[:word:]][[[:word:]]:.-]*\Z/
```

Comment:
The `[[` / `]]` brackets look weird, but are correct - see https://ruby-doc.org/core-2.1.1/Regexp.html#class-Regexp-label-Character+Classes

I can give `:alnum:` a shot with some test data to see what happens.  We're mostly concerned with making tags compatible with the concat module for https://tickets.puppetlabs.com/browse/MODULES-4864.